### ISSUE TITLE
[RFC] Improve gate-regime coverage for gated delta rule tests

### ISSUE THREAD

**Issue by Erix025** [fla-org/flash-linear-attention#891]
## Motivation

While debugging #889 and working on #890, I found that the issue was easier to reproduce when the gate was set to zero. This made the error much more visible because the relevant accumulated contribution was no longer attenuated by a small gate-dependent factor.

However, I think this points to a broader testing problem rather than only a `dg`-specific issue.

Currently, the gate in the tests is generated through a `logsigmoid`-style path. Since `logsigmoid(x)` is always non-positive, the cumulative gate value inside each chunk can quickly become very negative. As a result, the chunk-level `g_last` after `cumsum` is often very small in the exponential domain, i.e. `exp(g_last)` or `exp2(g_last)` can be close to zero.

This can strongly attenuate the contribution of recurrent states and accumulated gradients at chunk boundaries. Therefore, some bugs may be hidden simply because the sampled gates make the problematic path numerically insignificant.

In other words, passing the current tests does not necessarily mean that all gate-related paths are correct. Some errors may simply be hidden because the current logsigmoid-based gate distribution makes the corresponding contributions too small to be observed.

## Proposal

I propose making gate-regime coverage explicit in the gated delta test suite.

For example, we can add representative gate settings such as:

```python
# Case 1: zero gate / no decay.
# This keeps the recurrent state from being decayed and can make accumulated errors more visible.
g = torch.zeros(B, T, H, dtype=torch.float32)

# Case 2: weak decay.
# This covers a lightly decayed regime where errors are still relatively observable.
g = torch.full((B, T, H), -0.1, dtype=torch.float32)

# Case 3: stronger decay.
# This covers a more heavily decayed regime and checks correctness when the state is strongly attenuated.
g = torch.full((B, T, H), -1.0, dtype=torch.float32)

# Case 4: current random logsigmoid gate.
# This keeps the existing random-gate distribution for general coverage.
g = F.logsigmoid(torch.randn(B, T, H, dtype=torch.float32))
```

The goal is not only to protect the specific path fixed in #890, but to make the overall test suite more robust across gate-dependent numerical regimes.

## Expected benefit

This would make the tests less dependent on accidental properties of the sampled gate values and improve the chance of exposing bugs in different kernel paths.

In particular, explicit gate-regime tests could help check that:

- forward outputs remain correct across different decay strengths;
- gradients remain correct across different gate regimes;
- repeated runs with identical inputs are deterministic;
- errors are not hidden simply because the current `logsigmoid`-based gate distribution suppresses the affected path;
- future changes to gated delta kernels are tested against a broader set of numerical conditions.

## Relation to #889 / #890

#889 exposed a shared-memory race in the `chunk_bwd` path. #890 adds a regression test using `g = 0`, which helps catch that specific issue.

This RFC is about generalizing that lesson. Instead of treating `g = 0` only as a regression case for one bug, we may want to explicitly include several gate regimes in the gated delta tests so that the test suite can cover more numerical scenarios and potentially expose other hidden issues.

I am happy to help add these cases if this direction sounds reasonable.
